src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractRoslynProject.cs

```diff
@@ -24,8 +29,13 @@ internal abstract partial class AbstractRoslynProject : AbstractProject, ICompil
             string language,
             IServiceProvider serviceProvider,
             VisualStudioWorkspaceImpl visualStudioWorkspaceOpt,
-            HostDiagnosticUpdateSource hostDiagnosticUpdateSourceOpt)
-            : base(projectTracker, reportExternalErrorCreatorOpt, projectSystemName, hierarchy, language, serviceProvider, visualStudioWorkspaceOpt, hostDiagnosticUpdateSourceOpt)
+            HostDiagnosticUpdateSource hostDiagnosticUpdateSourceOpt,
+            string projectFilePath = null,
```

Comment:
Actually, I'm leaning towards that this is the wrong approach, AbstractProject is doing a bunch of work that expects its on the UI thread.